sklearn/feature_extraction/_dict_vectorizer.py

```diff
@@ -386,6 +391,25 @@ def get_feature_names(self):
         """
         return self.feature_names_
 
+    def get_feature_names_out(self, input_features=None):
+        """Get output feature names for transformation.
+
+        Parameters
+        ----------
+        input_features : array-like of str or None, default=None
+            Not used, present here for API consistency by convention.
```

Comment:
This could actually be useful to implement as a way to rename the columns after a complex pipeline was fit without refitting. But maybe we should consider that for an follow-up PR to keep this PR easy to review.